pkg/apiserver/apiserverconfig/longrunning.go

```diff
@@ -2,25 +2,36 @@ package apiserverconfig
 
 import (
 	"net/http"
-	"regexp"
 
 	"k8s.io/apimachinery/pkg/util/sets"
 	apirequest "k8s.io/apiserver/pkg/endpoints/request"
 	genericfilters "k8s.io/apiserver/pkg/server/filters"
-)
 
-// request paths that match this regular expression will be treated as long running
-// and not subjected to the default server timeout.
-const originLongRunningEndpointsRE = "(/|^)(buildconfigs/.*/instantiatebinary|imagestreamimports)$"
+	buildv1 "github.com/openshift/api/build/v1"
+	imagev1 "github.com/openshift/api/image/v1"
+)
 
 var (
-	originLongRunningRequestRE = regexp.MustCompile(originLongRunningEndpointsRE)
-	kubeLongRunningFunc        = genericfilters.BasicLongRunningRequestCheck(
+	kubeLongRunningFunc = genericfilters.BasicLongRunningRequestCheck(
 		sets.NewString("watch", "proxy"),
 		sets.NewString("attach", "exec", "proxy", "log", "portforward"),
 	)
 )
 
 func IsLongRunningRequest(r *http.Request, requestInfo *apirequest.RequestInfo) bool {
-	return originLongRunningRequestRE.MatchString(r.URL.Path) || kubeLongRunningFunc(r, requestInfo)
+	if requestInfo.APIGroup == buildv1.GroupName &&
+		requestInfo.APIVersion == buildv1.GroupVersion.Version &&
```

Comment:
do we have to check the api version ?
i don't think it is required in this context. 
skipping the version check would make this code forward compatible.

Reply:
Good idea, all of these requests are longrunning, regardless of the version